api/src/main/java/org/apache/iceberg/metrics/Histogram.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+public interface Histogram {
+  /** Update the histogram with a new value observed. */
+  void update(long value);
```

Comment:
I don't see a reason why this needs to be a `long`. It could easily be any primitive number, right?

Reply:
I think you are right. we can make it a generic type as `Histogram<T extends Number>`. then we just need to change `min` and `max` return type to `T`. everything else (like `mean` and `percentile`) can stay with `double` return type.

Reply:
while trying the generic type, I realized there are a couple of small performance implications
1) we will need to define impl classes like `IntHistogram`, `LongHistogram` and `DoubleHistogram`, as we can avoid create `T[]` using the wrapper/box class. the primitive type array is 50% smaller.
2) there will be unboxing overhead.

Reply:
Let's just leave it as long for now.